### PR TITLE
Cantor-Schröder-Bernstein

### PR DESCRIPTION
**Pull request by ElifUskuplu**
I wanted to add the proof of my favorite theorem, which is Cantor-Schröder-Bernstein, as a first contribution to the library. Martin Escardo proved it in HoTT and I just translated it into the language of this library. Hoping to contribute more. :) 

Probably, it needs improvement. I'm open to any suggestions. Thanks in advance.

### CODE REVIEW

**Comment by jonaprieto** [comment]
Awesome! Thank you!
Before reviewing your contribution in detail. Could you please fix it following the [style conventions of the library](https://github.com/UniMath/agda-unimath#indentation)? See, for example, how we indent after the colon in a type sig.

**Comment by ElifUskuplu** [comment]
Thanks, I will fix it accordingly.

**Comment by ElifUskuplu** [comment]
I've fixed. I'm not sure about its folder. That's why I added it to src folder directly.

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# Cantor-Schröder-Bernstein
+
+```agda
+{-# OPTIONS --without-K --exact-split #-}
+
+module Cantor-Schröder-Bernstein where
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:6]: It's better to put this module in a new folder called "set-theory".

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+open import foundation.propositional-maps using (is-prop-map-is-emb)
+open import foundation.retractions using (retr)
+open import foundation.sections using (sec)
+open import foundation.split-surjective-maps using (is-split-surjective)
+open import foundation.universe-levels using (Level; UU; _⊔_; lsuc)
+```
+
+## Idea
+
+The classical Cantor–Schröder–Bernstein Theorem of set theory, formulated by Cantor and first proved by Bernstein, states that for any pair of sets A and B, if there are injective maps f : A → B and g : B → A, then one can find a bijection h : A → B. There are proofs that use the principle of excluded middle but not the axiom of choice. That the principle excluded middle is absolutely necessary.
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:37]: You can use LaTeX here, e.g. $f : A \to B$.

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+Here, we will prove the following theorem assuming the terminology and notation of the HoTT book.
+
+From given embeddings of two types into each other, we can construct an equivalence between them using the principle of excluded middle.
+
+The idea and the proof is given by Martin Escardo in his paper "The Cantor–Schröder–Bernstein Theorem for ∞-groupoids". We will follow its proof using agda.unimath library settings.
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:43]: Better to link the paper  using Markdown:`["The Cantor–Schröder–Bernstein Theorem for ∞-groupoids"](https://doi.org/10.1007/s40062-021-00284-6)`

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+For any map f : A → A, we can define its iterated compositions. It's needed due to define `is-g-point` notion in the proof.
+
+```agda
+_^_ :  {l1 : Level} {A : UU l1} → (A → A) → ℕ → (A → A)
+f ^ zero-ℕ = id
+f ^ succ-ℕ n = f ∘ (f ^ n)
+```
+
+We also need an non-dependent version of induction principle for coproduct types
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:58]: It's not really needed. It's a particular case of `ind-prod`. Wherever you have `non-dep-ind-coprod`, you can simply replace it by `ind-coprod _`. You need to update the module `foundation.coproduct-types`

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+```agda
+non-dep-ind-coprod :
+  {l1 l2 l3 : Level} {A : UU l1} {B : UU l2 } {C : UU l3}
+  → (A → C) → (B → C)
+  → (coprod A B → C)
+non-dep-ind-coprod f g (inl x) = f x
+non-dep-ind-coprod f g (inr y) = g y
+```
+
+It's useful to prove that any injective split-surjection is an equivalence.
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:69]: It is useful. Can you move this to `foundation.split-surjective-maps` in a new section called `Properties`.

File: src/Cantor-Schröder-Bernstein.lagda.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+f ^ zero-ℕ = id
+f ^ succ-ℕ n = f ∘ (f ^ n)
+```
+
+We also need an non-dependent version of induction principle for coproduct types
+
+```agda
+non-dep-ind-coprod :
+  {l1 l2 l3 : Level} {A : UU l1} {B : UU l2 } {C : UU l3}
+  → (A → C) → (B → C)
```
Comment (jonaprieto) [inline src/Cantor-Schröder-Bernstein.lagda.md:63]: Btw, we don't indent using the arrows, they must be at the end of the lines.

**Comment by ElifUskuplu** [comment]
I've made changes and a new pull-request.

**Pull request closed without merging**